File: Backend/LibraryMS/Lib.Domain/Entities/Book.cs
namespace Lib.Domain.Entities;

public class Book
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string ISBN { get; set; } = string.Empty;
    public DateTime PublishedDate { get; set; }
    public int Quantity { get; set; }
    public int Available { get; set; }
    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public ICollection<BookBorrowingRequestDetails> BorrowingDetails { get; set; } = [];
}

File: Frontend/LibraryMS/src/models/Book.ts
export interface Book {
  id: string;
  title: string;
  author: string;
  description: string;
  isbn: string;
  publishedDate: string;
  categoryId: string;
  categoryName: string;
  quantity: number;
  available: number;
}

export interface BookAPI {
  items: Book[];
  pageNumber: number;
  pageSize: number;
  totalItems: number;
  totalPages: number;
}